### PR TITLE
Separate ghost warp message into two

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
Previously it was setting one of the two properties to differentiate between warps to warp locations and entities, now it uses 2 separate messages instead.

### CODE REVIEW

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    [Serializable, NetSerializable]
+    public class GhostWarpToNameRequestMessage : ComponentMessage
+    {
+        public GhostWarpToNameRequestMessage(string warpTarget = default)
+        {
+            WarpName = warpTarget;
+            Directed = true;
+        }
+
+        public string WarpName { get; }
```
Comment (Visne) [inline Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs:16]: I feel like it's more clean to put this above the constructor.

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToTargetRequestMessage.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    [Serializable, NetSerializable]
+    public class GhostWarpToTargetRequestMessage : ComponentMessage
+    {
+        public GhostWarpToTargetRequestMessage(EntityUid target)
+        {
+            Target = target;
+            Directed = true;
+        }
+
+        public EntityUid Target { get; }
```
Comment (Visne) [inline Content.Shared/GameObjects/Components/Observer/GhostWarpToTargetRequestMessage.cs:16]: Same thing.

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.GameObjects.Components.Observer
+{
+    [Serializable, NetSerializable]
+    public class GhostWarpToNameRequestMessage : ComponentMessage
+    {
+        public GhostWarpToNameRequestMessage(string warpTarget = default)
```
Comment (Visne) [inline Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs:10]: 1. I suggest renaming `warpTarget` to `name`.
2. Is it still required to put `= default` here?

File: Content.Server/GameObjects/Components/Observer/GhostComponent.cs [hunk 4]
```diff
@@ -94,43 +94,48 @@ public override void HandleNetworkMessage(ComponentMessage message, INetChannel
...
 
-                    if (warp.PlayerTarget != default)
+                    foreach (var warpPoint in FindWaypoints())
                     {
-                        if (!Owner.EntityManager.TryGetEntity(warp.PlayerTarget, out var entity))
+                        if (warp.WarpName == warpPoint.Location)
                         {
+                            Owner.Transform.Coordinates = warpPoint.Owner.Transform.Coordinates;
                             break;
                         }
```
Comment (Visne) [inline Content.Server/GameObjects/Components/Observer/GhostComponent.cs:110]: Maybe log a warning here if a user tries to warp to a nonexistant warp location?

**Review by Visne — APPROVED** [review]
Some suggestions.

File: Content.Server/GameObjects/Components/Observer/GhostComponent.cs (hunk 4 above)
Comment (DrSmugleaf) [reply Content.Server/GameObjects/Components/Observer/GhostComponent.cs:111]: Done in https://github.com/space-wizards/space-station-14/pull/3310/commits/141de1126f31e66cc5db5d954ab65aadc3fbf794

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs (hunk 3 above)
Comment (DrSmugleaf) [reply Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs]: Done and fixed in https://github.com/space-wizards/space-station-14/pull/3310/commits/141de1126f31e66cc5db5d954ab65aadc3fbf794

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Shared/GameObjects/Components/Observer/GhostWarpToNameRequestMessage.cs]: Done in https://github.com/space-wizards/space-station-14/pull/3310/commits/712431c27a251123feeed82cf066580a24dce43e

File: Content.Shared/GameObjects/Components/Observer/GhostWarpToTargetRequestMessage.cs (hunk 2 above)
Comment (DrSmugleaf) [reply Content.Shared/GameObjects/Components/Observer/GhostWarpToTargetRequestMessage.cs]: Done in https://github.com/space-wizards/space-station-14/pull/3310/commits/712431c27a251123feeed82cf066580a24dce43e

**Review by Visne — APPROVED** [review]

**Pull request merged**